contracts/gateway/EIP20Gateway.sol

```diff
@@ -923,13 +923,13 @@ contract EIP20Gateway is GatewayBase {
             storageRoot
         );
 
-        // delete the unstake data
-        delete unstakes[_messageHash];
-
         redeemer_ = message.sender;
         redeemerNonce_ = message.nonce;
         amount_ = unstakes[_messageHash].amount;
 
+        // delete the unstake data
+        delete unstakes[_messageHash];
```

Comment:
After progress and revert you may not be able to do anything with stakes/unstakes mapping else so its deleted. Message state is not deleted because it may be used for merkle proof.

But in fact, I am thinking not to delete the message state.


1. Stake
2. ConfirmStake
3. ProgressMintWithProof. (deletes the mints stuct data, the proof was not generated)
4. redeem (deletes the message status)
5. ProgressStakeWithProof (Now the proof will not be generated.)

Reply:
I understand we can't clean message status mapping and it has valid reason.
My point was for message mapping. Messages can be deleted with stake, mint, redeem  and unstakes mapping. 

If you and @schemar  agree on this, seperate ticket can be created for this.